samples/dnn/mask_rcnn.cpp

```diff
@@ -0,0 +1,198 @@
+    "{ image m     |<none>| Path to input image file.  }"
+    "{ video v     |<none>| Path to input video file.  }"
+    "{ classes n   |./mscoco_labels.names | Path to a text file with names of classes.  }"
+    "{ model w     |./mask_rcnn_inception_v2_coco_2018_01_28/frozen_inference_graph.pb | The pre-trained weights.  }"
+    "{ config c    |./mask_rcnn_inception_v2_coco_2018_01_28.pbtxt | iThe text graph file that has been tuned by the OpenCV’s DNN support group  }"
+    "{ cthr        | .5 | Confidence threshold. }"
+    "{ mthr        | .4 | Mask threshold. }";
+
+void drawBox(Mat& frame, int classId, float conf, Rect box, Mat& objectMask, float maskThreshold, vector<Scalar> colors, vector<string> classes);
+
+void postprocess(Mat& frame, const vector<Mat>& outs, float confThreshold, float maskThreshold, vector<Scalar> colors, vector<string> classes);
+
+int main(int argc, char** argv)
+{
+    CommandLineParser parser(argc, argv, keys);
+    parser.about("This sample demonstrates instance segmentation network called Mask-RCNN.");
+    if (argc == 1 || parser.has("help"))
+    {
+        parser.printMessage();
+        return 0;
+    }
+
+    float confThreshold;
+    float maskThreshold;
+    vector<string> classes;
+    vector<Scalar> colors;
+
+
+    confThreshold = parser.get<float>("cthr");
+    maskThreshold = parser.get<float>("mthr");
+
+    String textGraph = parser.get<string>("config");
+    String modelWeights = parser.get<string>("model");
+    string classesFile = parser.get<string>("classes");
+
+    ifstream ifs(classesFile.c_str());
+    string line;
+    while (getline(ifs, line)) classes.push_back(line);
+
+    for(int _c =0; _c<=10; _c++)
+    {
+        colors.push_back( Scalar(rand() % 255,rand() % 255,rand() % 255 ,255.0));
+    }
+
+    Net net = readNet(modelWeights, textGraph,"tensorflow");
+
+    string str;
+    VideoCapture cap;
+    Mat frame, blob;
+
+    try {
+
+        if (parser.has("image"))
+        {
+            str = parser.get<String>("image");
+            cout << "Image file input : " << str << endl;
+            ifstream ifile(str);
+            if (!ifile) throw("error");
+            cap.open(str);
+        }
+        else if (parser.has("video"))
+        {
+            str = parser.get<String>("video");
+            ifstream ifile(str);
+            if (!ifile) throw("error");
+            cap.open(str);
+        }
+        else cap.open(parser.get<int>("device"));
+
+    }
+    catch(...) {
+        cout << "Could not open the input image/video stream" << endl;
+        return 0;
+    }
+
+    while (waitKey(1) < 0)
+    {
+        cap >> frame;
+
+        if (frame.empty()) {
+            cout << "Done processing !!!" << endl;
+            waitKey(0);
+            break;
+        }
+
+        blobFromImage(frame, blob, 1.0, Size(frame.cols, frame.rows), Scalar(), true, false);
```

Comment:
There are should be a command line argument to specify input blob dimensions. Take a look onto https://github.com/opencv/opencv/blob/master/samples/dnn/mask_rcnn.py.

Reply:
ok, done